### PR TITLE
Feature/sdev 3256 gene list updates pharma cancer predisp

### PR DESCRIPTION
**Pull request by dustinbleile**
Migrate get_cancer_predisposition_info and get_pharmacogenomic_info from gsc_report added to genes.py

These functions use a preferred gene_name to replace transcripts, alternated representations.  Added an equivalent function to genes.py, get_preferred_gene_name.

### CODE REVIEW

File: graphkb/constants.py [hunk 1]
```diff
@@ -56,8 +56,10 @@
 ONCOKB_SOURCE_NAME = 'oncokb'
 ONCOGENE = 'oncogenic'
 TUMOUR_SUPPRESSIVE = 'tumour suppressive'
-
 FUSION_NAMES = ['structural variant', 'fusion']
+PHARMACOGENOMIC_RELEVANCE_TERMS = ["decreased toxicity", "increased toxicity"]
```
Comment (dustinbleile) [inline graphkb/constants.py:60]: Hi @mattdoug604 , this constant was from gsc_report.

From the RELEVANCE_BASE_TERMS constant below, it seems like pharmacogenomic relevance terms should include metabolism and dosage.

`('pharmacogenomic', ['metabolism', 'toxicity', 'dosage']),`

Does that seem right, in general?  (There were probably just no metabolism or dosage GKB variants to count yet?)
Details on https://www.bcgsc.ca/jira/browse/SDEV-2733 might be relevant.

**Comment by dustinbleile** [comment]
Updated

File: graphkb/genes.py [hunk 2]
```diff
@@ -161,3 +162,200 @@ def get_genes_from_variant_types(
...
+        ignore_cache (bool, optional): bypass the cache to always force a new request
+        source: id of the preferred gene symbol source
+    Returns:
+        preferred displayName symbol.
+
+    Example:
+        return KRAS for get_preferred_gene_name(conn, 'NM_033360')
+        return KRAS for get_preferred_gene_name(conn, 'ENSG00000133703.11')
+    """
+    CHROMOSOMES = [f"chr{i}" for i in range(1, 24)] + ['chrX', 'chrY']
```
Comment (mattdoug604) [inline graphkb/genes.py:181]: might make sense to move this to `constants.py`

Comment (mattdoug604) [reply graphkb/genes.py:181]: Also `x`, `y`, and `mt` are references used in GraphKB. What happens if one of those is passed?

File: graphkb/genes.py [hunk 3]
```diff
@@ -161,3 +162,200 @@ def get_genes_from_variant_types(
...
+    if gene_name in CHROMOSOMES:
+        logger.error(f"{gene_name} assumed to be a chromosome, not gene")
+        return ''
+    eq = get_equivalent_features(conn=conn, gene_name=gene_name)
+    genes = [m for m in eq if m.get('biotype', '') == 'gene' and not m.get('deprecated', False)]
```
Comment (mattdoug604) [inline graphkb/genes.py:186]: minor, `m.get('biotype', '')` can just be `m.get('biotype')`

File: graphkb/genes.py [hunk 4]
```diff
@@ -161,3 +162,200 @@ def get_genes_from_variant_types(
...
+    if not genes:
+        logger.error(f"No genes found for: {gene_name}")
+        return ''
+    if source:
+        source_filtered_genes = [m for m in genes if m.get('source', '') == source]
```
Comment (mattdoug604) [inline graphkb/genes.py:191]: minor, `m.get('source', '')` can just be `m.get('source')`

File: graphkb/genes.py [hunk 5]
```diff
@@ -161,3 +162,200 @@ def get_genes_from_variant_types(
         ),
     )
     return result
+
+
+def get_preferred_gene_name(conn: GraphKBConnection, gene_name: str, source: str = '#39:5') -> str:
```
Comment (mattdoug604) [inline graphkb/genes.py:167]: It'd be helpful to move `'#39:5'` to constants and add a comment about what it is. Something like `PREFERRED_GENE_SOURCE = "#39.5"  # HGNC`

File: graphkb/genes.py (hunk 2 above)
Comment (dustinbleile) [reply graphkb/genes.py:181]: > Also x, y, and mt are references used in GraphKB. What happens if one of those is passed?

Currently, they would be mistaken for gene names.

I think its a good idea to move this to constants.py.  Will do.

**Comment by dustinbleile** [comment]
Ping - new reviewers

File: graphkb/genes.py (hunk 3 above)
Comment (dustinbleile) [reply graphkb/genes.py]: Done

**Comment by dustinbleile** [comment]
Hi @calchoo & @darenillas

**Comment by dustinbleile** [comment]
Hi @darenillas, @calchoo, @mattdoug604 

pinging for some review

**Review by calchoo — APPROVED** [review]

**Pull request merged**